src/com/group6/users/UserStorage.java

```diff
@@ -0,0 +1,117 @@
+import java.io.*;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Handles the storage and retrieval of user data from text files.
+ * <p>
+ * This class loads users from role-specific files and provides methods
+ * to save user data persistently. It supports Applicants, Officers, and Managers.
+ * </p>
+ */
+public class UserStorage {
+    private final String applicantsFilepath;
+    private final String officersFilepath;
+    private final String managersFilepath;
+
+    /**
+     * Constructs a UserStorage instance with specified file paths.
+     *
+     * @param applicantsFilepath The file path for storing applicant data.
+     * @param officersFilepath   The file path for storing officer data.
+     * @param managersFilepath   The file path for storing manager data.
+     */
+    public UserStorage(String applicantsFilepath, String officersFilepath, String managersFilepath) {
+        this.applicantsFilepath = applicantsFilepath;
+        this.officersFilepath = officersFilepath;
+        this.managersFilepath = managersFilepath;
+    }
+
+    /**
+     * Loads all users from the respective files.
+     *
+     * @return A map of users keyed by NRIC.
+     */
+    public Map<String, User> loadAllUsers() {
+        Map<String, User> users = new HashMap<>();
+        loadUsersFromFile(users, applicantsFilepath, UserRole.APPLICANT);
+        loadUsersFromFile(users, officersFilepath, UserRole.OFFICER);
+        loadUsersFromFile(users, managersFilepath, UserRole.MANAGER);
+        return users;
+    }
+
+    /**
+     * Reads user data from the specified file and adds them to the provided map.
+     *
+     * @param users    The map where loaded users will be stored.
+     * @param filename The file from which users will be loaded.
+     * @param role     The role of the users being loaded (Applicant, Officer, or Manager).
+     */
+    private void loadUsersFromFile(Map<String, User> users, String filename, UserRole role) {
+        try (BufferedReader reader = new BufferedReader(new FileReader(filename))) {
+            String line;
+            while ((line = reader.readLine()) != null) {
+                String[] parts = line.split(",");
+                if (parts.length == 5) {
+                    User user = createUserByRole(parts, role);
+                    users.put(user.getNric(), user);
+                }
+            }
+        } catch (IOException e) {
+            System.out.println("Error reading from file: " + filename);
+        }
+    }
+
+    /**
+     * Creates a User object based on the given role and user data.
+     *
+     * @param parts The array containing user attributes (name, NRIC, age, marital status, password).
+     * @param role  The role of the user (Applicant, Officer, or Manager).
+     * @return A new User object corresponding to the provided role.
+     */
+    private static User createUserByRole(String[] parts, UserRole role) {

```

Comment:
Consider adding more safety checks with parsing. 

Here's smth I can do that will break this:
Say `applicants.csv`
```
asdf, asdf, NOT_A_NUMBER
```
This will cause the entire app to crash because `Integer.parseInt` will parse `NOT_A_NUMBER`.

Do the following checks:
- Length of `parts` matches expected.
- Integer parsing in a try catch (NumberFormatException)

If theres an error for this row, then "skip it" but provide a warning